### ISSUE TITLE
Allow disabling of bundleDir from plugin option

### ISSUE THREAD

**Issue by amoshydra** [webpack-contrib/webpack-bundle-analyzer#274]
#### Issue description
Allow us to disable/overwrite `bundleDir` in the plugin option

Currently `bundleDir` can only be defined in via the command line. In plugin `bundleDir` is decided in the following logic

https://github.com/webpack-contrib/webpack-bundle-analyzer/blob/1c8aba35c62850a2e30a978471efba8f3925b643/src/BundleAnalyzerPlugin.js#L130-L132

However, `compiler.outputFileSystem` might not always be `MemoryFileSystem`. 

##### Use case:
The framework `Nuxt 2.5.0` [replace](https://github.com/nuxt/nuxt.js/blob/fbabeb8fce4630c193416e178f10f75d841083b5/packages/webpack/src/utils/async-mfs.js) `MemoryFileSystem` with their own implementation of `AsyncMFS`, which extend `MemoryFileSystem`, during development.

##### Current problem
This cause `compiler.outputFileSystem.constructor.name` to be `"AsyncMFS"`, thus causing`getBundleDirFromCompiler` to return `this.compiler.outputPath`. 

##### Expected behavior
The expectation result here is `null`. Since nuxt is using `webpack-dev-server` during development and no real file is being created.

##### My thoughts
I am not sure if it is possible to detect if real file is created in this case. It might not be reasonable for `webpack-bundle-analyzer` to keep track of a list of `outputFileSystem.constructor.name` that should make `bundleDir` to `null`.

Thus I am proposing `webpack-bundle-analyzer` to allow us to overwrite `bundleDir` to `null` in the plugin option

#### Technical info
* Webpack Bundle Analyzer version: v3.3.2
* Webpack version: v4.30.0
* Node.js version: v8.11.3
* npm/yarn version: v1.12.1
* OS: Windows 10, 1803 (OS Build 17134.706)

#### Debug info
How do you use this module? As CLI utility or as plugin?

> As plugin

If CLI, what command was used? (e.g. `webpack-bundle-analyzer -O path/to/stats.json`)

If plugin, what options were provided? (e.g. `new BundleAnalyzerPlugin({ analyzerMode: 'disabled', generateStatsFile: true })`)

```
new webpackBundleAnalyzer.BundleAnalyzerPlugin({
  openAnalyzer: false,
}),
```
What other Webpack plugins were used?

> This plugin is used with https://github.com/nuxt/nuxt.js `v2.6.2`

It would be nice to also attach webpack stats file.
It can be generated using these options:
```js
new BundleAnalyzerPlugin({
  analyzerMode: 'disabled',
  generateStatsFile: true,
  // Excludes module sources from stats file so there won't be any sensitive data
  statsOptions: { source: false }
})
```
`stats.json` will be created in Webpack bundle output directory.

> not available due to private content in file path and name

**Comment by valscion** [comment]
The simplest way to get Nuxt unblocked could be to amend that conditional to also list `AsyncMFS` and then add a comment above to describe why that check is necessary, preferrably linking also to this issue.

Would you be open to create such a PR?

**Comment by amoshydra** [comment]
Hi @valscion, thanks for the prompt response.

I have create 2 pull requests for this issue
- https://github.com/webpack-contrib/webpack-bundle-analyzer/pull/276 Add AsyncMFS for getBundleDirFromCompiler
- https://github.com/webpack-contrib/webpack-bundle-analyzer/pull/275 Allow overwriting of `bundleDir` via `bundleDirOption` from plugin option

Would like to have your opinion on which PR is more suitable for `webpack-bundle-analyzer`.

**Comment by valscion** [comment]
Let's go with #276 first as I'm a bit wary on adding new configuration options for something that seems to be quite an edge case. I'd like us to see more concrete use cases than Nuxt before adding a new config option

**Comment by th0r** [comment]
Fixed in #275
